src/include/planner/plannodes/abstract_plan_node.h

```diff
@@ -84,7 +84,7 @@ class AbstractPlanNode {
   /**
    * @return child plan nodes
    */
-  const std::vector<std::shared_ptr<AbstractPlanNode>> &GetChildren() const { return children_; }
+  const std::vector<std::unique_ptr<AbstractPlanNode>> &GetChildren() const { return children_; }
```

Comment:
I feel like this should be `std::vector<std::manged_ptr<AbstractPlanNode>>`. If this function is giving out a reference to this vector and let's say someone makes a deep copy of the vector, how would unique_ptrs work in that case? Would there be two unique_ptrs pointing to the same location in memory? And when that  copy unique_ptr goes out of scope is memory it's pointing to freed? Wouldn't that be bad since data would be freed while abstract_plan_node is still using it?

Reply:
Yep